source/client/process_impl.h

```diff
@@ -107,5 +107,18 @@ class ProcessImpl : public Process, public Envoy::Logger::Loggable<Envoy::Logger
   bool shutdown_{true};
 };
 
+/**
+ * Will delegate execution to a remote nighthawk_service using gRPC.
+ */
+class RemoteProcessImpl : public Process, public Envoy::Logger::Loggable<Envoy::Logger::Id::main> {
```

Comment:
(optional) Would this work better in a new file called say remote_process.[h|cc]?

Reply:
Done: 8b32683c8b89cf8060237eaa28100e7eb7bb7123